frontend/src/app/commonComponents/RadioGroup.tsx

```diff
@@ -63,94 +65,109 @@ const RadioGroup = <T extends string>({
+                  <label className={labelClasses} htmlFor={uid(c.value)}>
+                    {c.label}
+                    {c.labelDescription && (
+                      <span className="usa-checkbox__label-description text-base-dark">
+                        {c.labelDescription}
+                      </span>
+                    )}
+                    {c.labelTag && (
+                      <div className="display-block margin-top-1">
+                        <span className="usa-tag bg-primary-darker">
+                          {c.labelTag}
+                        </span>
+                      </div>
+                    )}
+                  </label>
+                </div>
+              );
+            })}
           </div>
         )}
-        <UIDConsumer>
-          {(_, uid) => (
-            <div
-              className={classnames(
-                variant === "horizontal" && "prime-radio--horizontal"
-              )}
-              {...(validationStatus === "error"
-                ? { "aria-describedby": `error_${name}`, "aria-invalid": true }
-                : null)}
-            >
-              {buttons.map((c) => {
-                const labelClasses = classnames(
-                  "usa-radio__label",
-                  (c.disabled || disabled) && "text-base"
-                );
-                const className = classnames(groupClass, c.className);
-                return (
-                  <div className={className} key={uid(c.value)}>
-                    <input
-                      type="radio"
-                      id={uid(c.value)}
-                      name={name}
-                      value={c.value}
-                      data-required={required || "false"}
-                      disabled={disabled || c.disabled || false}
-                      className={inputClass}
-                      checked={c.value === selectedRadio}
-                      onClick={onClick ? () => onClick(c.value) : undefined}
-                      onChange={() => onChange(c.value)}
-                      onBlur={onBlur}
-                    />
-                    <label className={labelClasses} htmlFor={uid(c.value)}>
-                      {c.label}
-                      {c.labelDescription && (
-                        <span className="usa-checkbox__label-description text-base-dark">
-                          {c.labelDescription}
-                        </span>
-                      )}
-                      {c.labelTag && (
-                        <div className="display-block margin-top-1">
-                          <span className="usa-tag bg-primary-darker">
-                            {c.labelTag}
-                          </span>
-                        </div>
-                      )}
-                    </label>
-                  </div>
-                );
-              })}
-            </div>
+      </UIDConsumer>
+    </fieldset>
+  );
+
+  return (
+    <>
+      {renderAsForm ? (
```

Comment:
if the only logic changing in this conditional block is the tag then an approach like the one in [this PR](https://github.com/CDCgov/prime-simplereport/pull/5371/files) with the headings would be cleaner